internal/pkg/agent/application/coordinator/coordinator_test.go

```diff
@@ -313,6 +314,78 @@ func TestCoordinator_StateSubscribe(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestCoordinatorWithErrors(t *testing.T) {
+	handlerChan, runtime, varWatcher, config := setupAndWaitCoordinatorDone()
+
+	close(runtime)
+	close(varWatcher)
+	cfgErrStr := "configWatcher error"
+	config <- errors.New(cfgErrStr)
+
+	// All the multierror stuff breaks errors.Is
+	waitAndTestError(t, func(err error) bool { return strings.Contains(err.Error(), cfgErrStr) }, handlerChan)
+
+}
+
+func TestCoordinatorShutdownClosedChannels(t *testing.T) {
```

Comment:
You should test the case where they all return an error without any of them closing.